### PR TITLE
Dan/read source data

### PR DESCRIPTION
**Pull request by DanPaseltiner**
This PR adds an event triggered Cloud Function to read source data that should be run through the ingestion pipeline. It publishes every message contained in source files to a gcp pubsub topic. Eventually our Workflow that orchestrates the Ingestion Pipeline will subscribe to this topic.

### CODE REVIEW

**Comment by celestemespinoza** [comment]
Task linked: [CU-3bukk3q Create Cloud Function for handling batch HL7/CCDA inputs](https://app.clickup.com/t/3bukk3q)

**Comment by celestemespinoza** [comment]
Task linked: [CU-2q4uz5t Make GCP Storage Buckets for Ingestion Pipeline](https://app.clickup.com/t/2q4uz5t)

File: cloud-functions/read_source_data/main.py [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+
+    # Read file.
+    storage_client = storage.Client()
+    bucket = storage_client.get_bucket(bucket_name)
+    blob = bucket.blob(filename)
+    file_contents = blob.download_as_text(encoding="utf-8")
+
+    # Handle batch Hl7v2 messages.
+    if message_type == "hl7v2":
+        messages = convert_batch_messages_to_list(file_contents)
```
Comment (nickclyde) [inline cloud-functions/read_source_data/main.py:96]: 💯

File: cloud-functions/read_source_data/test_read_source_data.py [hunk 2]
```diff
@@ -0,0 +1,242 @@
...
+    assert (
+        response.response
+        == log_info_and_generate_response(
+            (
+                "Processed source-data/elr/some-filename.txt, which contained 1 "
+                "messages, of which 1 were successfully published, and 0 could not be "
+                "published."
+            ),
+            "200",
+        ).response
```
Comment (nickclyde) [inline cloud-functions/read_source_data/test_read_source_data.py:241]: wow, thanks for these super thorough tests!

File: terraform/modules/cloud-functions/main.tf [hunk 3]
```diff
@@ -21,3 +21,25 @@ resource "google_cloudfunctions_function" "upload-fhir-bundle" {
...
+resource "google_cloudfunctions_function" "read_source_data" {
+  name        = "phdi-${terraform.workspace}-read_source_data"
+  description = "Read source data from bucket and publish to pub/sub topic for ingestion."
+  runtime     = "python39"
+
+  available_memory_mb   = 128
+  source_archive_bucket = var.functions_storage_bucket
+  source_archive_object = var.read_source_data_source_zip
+  event_trigger {
+    event_type = "google.storage.object.finalize"
```
Comment (nickclyde) [inline terraform/modules/cloud-functions/main.tf:34]: Nice!

File: terraform/modules/storage/main.tf [hunk 4]
```diff
@@ -46,6 +55,23 @@ resource "google_storage_bucket_object" "upload_fhir_bundle_source_zip" {
 
   # Append to the MD5 checksum of the files's content
   # to force the zip to be updated as soon as a change occurs
-  name   = "src-${data.archive_file.upload_fhir_bundle.output_md5}-${var.project_id}.zip"
+  name   = "src-${terraform.workspace}-${data.archive_file.upload_fhir_bundle.output_md5}.zip"
```
Comment (nickclyde) [inline terraform/modules/storage/main.tf:58]: Good change 👍

**Review by nickclyde — APPROVED** [review]
Excellent work here, Dan. Can't wait to see it in action!

**Pull request merged**